substrate/frame/support/src/traits/dynamic_params.rs

```diff
@@ -0,0 +1,149 @@
+		<<Self as RuntimeParameterStore>::AggregratedKeyValue as AggregratedKeyValue>::AggregratedValue:
+			TryIntoKey<<KV as AggregratedKeyValue>::AggregratedValue>,
+		<KV as AggregratedKeyValue>::AggregratedValue: TryInto<K::WrappedValue>;
+}
+
+/// A dynamic parameter store across a concrete KV type.
+pub trait ParameterStore<KV: AggregratedKeyValue> {
+	/// Get the value of a parametrized key.
+	fn get<K>(key: K) -> Option<K::Value>
+	where
+		K: Key + Into<<KV as AggregratedKeyValue>::AggregratedKey>,
+		<KV as AggregratedKeyValue>::AggregratedValue: TryInto<K::WrappedValue>;
+}
+
+/// Key of a dynamic parameter.
+pub trait Key {
+	/// The value that the key is parametrized with.
+	type Value;
+
+	/// An opaque representation of `Self::Value`.
+	type WrappedValue: Into<Self::Value>;
+}
+
+/// The aggregated key-value type of a dynamic parameter store.
+pub trait AggregratedKeyValue: Parameter {
+	/// The aggregated key type.
+	type AggregratedKey: Parameter + MaxEncodedLen;
+
+	/// The aggregated value type.
+	type AggregratedValue: Parameter + MaxEncodedLen;
+
+	/// Split the aggregated key-value type into its parts.
+	fn into_parts(self) -> (Self::AggregratedKey, Option<Self::AggregratedValue>);
+}
+
+impl AggregratedKeyValue for () {
+	type AggregratedKey = ();
+	type AggregratedValue = ();
+
+	fn into_parts(self) -> (Self::AggregratedKey, Option<Self::AggregratedValue>) {
+		((), None)
+	}
+}
+
+/// Allows create a `ParameterStore` from a `RuntimeParameterStore`.
+///
+/// This concretization is useful when configuring pallets, since a pallet will require a parameter
+/// store for its own KV type and not the aggregated runtime-wide KV type.
+pub struct ParameterStoreAdapter<PS, KV>(sp_std::marker::PhantomData<(PS, KV)>);
+
+impl<PS, KV> ParameterStore<KV> for ParameterStoreAdapter<PS, KV>
+where
+	PS: RuntimeParameterStore,
+	KV: AggregratedKeyValue,
+	<KV as AggregratedKeyValue>::AggregratedKey:
+		IntoKey<<<PS as RuntimeParameterStore>::AggregratedKeyValue as AggregratedKeyValue>::AggregratedKey>,
+	<KV as AggregratedKeyValue>::AggregratedValue:
+		TryFromKey<<<PS as RuntimeParameterStore>::AggregratedKeyValue as AggregratedKeyValue>::AggregratedValue>,
+{
+	fn get<K>(key: K) -> Option<K::Value>
+	where
+		K: Key + Into<<KV as AggregratedKeyValue>::AggregratedKey>,
+		<KV as AggregratedKeyValue>::AggregratedValue: TryInto<K::WrappedValue>,
+	{
+		PS::get::<KV, K>(key)
+	}
+}
+
+// workaround for rust bug https://github.com/rust-lang/rust/issues/51445
+mod workaround {
+	pub trait FromKey<T>: Sized {
+		#[must_use]
+		fn from_key(value: T) -> Self;
+	}
+
+	pub trait IntoKey<T>: Sized {
+		#[must_use]
+		fn into_key(self) -> T;
+	}
+
+	impl<T, U> IntoKey<U> for T
+	where
+		U: FromKey<T>,
+	{
+		fn into_key(self) -> U {
+			U::from_key(self)
+		}
+	}
+
+	pub trait TryIntoKey<T>: Sized {
+		type Error;
+
+		fn try_into_key(self) -> Result<T, Self::Error>;
```

Comment:
Brian wrote it and originally named them `TryInto2`, `TryFrom2` ... and so on. I think its just for having multiple TryFrom/Into impls that dont conflict.

Reply:
yes, but I do not see them being implemented by all those type generated with dynamic params

Reply:
Ah yes, that indeed looks weird, since its only used by the [ORML macros](https://github.com/open-web3-stack/open-runtime-module-library/blob/91942bcc134c5f073b3198d2eb9ebdec48d8547f/traits/src/parameters.rs#L403).  
They also support parametrized-keys, but i think that would make it too complicated for now.